csharp/src/Ice/Internal/Instance.cs

```diff
@@ -1573,6 +1490,12 @@ private NetworkProxy createNetworkProxy(Ice.Properties props, int protocolSuppor
     private const int StateActive = 0;
     private const int StateDestroyInProgress = 1;
     private const int StateDestroyed = 2;
+
+    // The lazy initialization ensures GetAssemblies is only called the first time we decode a Slice class or exception,
```

Comment:
Possibly the most controversial change.

In the old code, AssemblyUtil would load and reload all assemblies in case more assemblies were loaded since the previous call. I don't see how this situation can happen in practice, and we definitely don't encounter this situation in the test suite.

Reply:
I think this behavior is fine.